src/components/Request/NewRequestForm.Wizard.tsx

```diff
@@ -0,0 +1,193 @@
+  },
+  prev: (_methods: string[]) => "HiringInfo",
+});
+
+INPUTSTEPS.set("JOA", {
+  next: (methods: string[]) => {
+    const linkedinPost = methods.includes("linkedinPost");
+    return linkedinPost
+      ? "LinkedInPost"
+      : INPUTSTEPS.get("LinkedInPost")?.next(methods) ?? "LinkedInPost";
+  },
+  prev: (methods: string[]) => {
+    const lcmc = methods.includes("lcmc");
+    return lcmc
+      ? "JobBoard"
+      : INPUTSTEPS.get("JobBoard")?.prev(methods) ?? "JobBoard";
+  },
+});
+
+INPUTSTEPS.set("LinkedInPost", {
+  next: (methods: string[]) => {
+    const linkedinSearch = methods.includes("linkedinSearch");
+    return linkedinSearch
+      ? "LinkedInSearch"
+      : INPUTSTEPS.get("LinkedInSearch")?.next(methods) ?? "LinkedInSearch";
+  },
+  prev: (methods: string[]) => {
+    const joa = methods.includes("joa");
+    return joa ? "JOA" : INPUTSTEPS.get("JOA")?.prev(methods) ?? "JOA";
+  },
+});
+
+INPUTSTEPS.set("LinkedInSearch", {
+  next: (methods: string[]) => {
+    const usaJobsFlyer = methods.includes("usaJobsFlyer");
+    return usaJobsFlyer
+      ? "USAJobs"
+      : INPUTSTEPS.get("USAJobs")?.next(methods) ?? "USAJobs";
+  },
+  prev: (methods: string[]) => {
+    const linkedinPost = methods.includes("linkedinPost");
+    return linkedinPost
+      ? "LinkedInPost"
+      : INPUTSTEPS.get("LinkedInPost")?.prev(methods) ?? "LinkedInPost";
+  },
+});
+
+INPUTSTEPS.set("USAJobs", {
+  next: (_methods: string[]) => "DONE",
+  prev: (methods: string[]) => {
+    const linkedinSearch = methods.includes("linkedinSearch");
+    return linkedinSearch
+      ? "LinkedInSearch"
+      : INPUTSTEPS.get("LinkedInSearch")?.prev(methods) ?? "LinkedInSearch";
+  },
+});
+
+INPUTSTEPS.set("DONE", {
+  next: (_methods: string[]) => "DONE",
+  prev: (methods: string[]) => {
+    const linkedinSearch = methods.includes("usaJobsFlyer");
+    return linkedinSearch
+      ? "USAJobs"
+      : INPUTSTEPS.get("USAJobs")?.prev(methods) ?? "USAJobs";
+  },
+});
+
+interface ActionType {
+  type: string;
+  payload?: string[];
+}
+
+function reducer(state: { page: string }, action: ActionType) {
+  switch (action.type) {
+    case "next_page":
+      const nextPage = INPUTSTEPS.get(state.page)?.next(action.payload ?? []);
+
+      return nextPage ? { ...state, page: nextPage } : state;
+
+    case "prev_page":
+      const prevPage = INPUTSTEPS.get(state.page)?.prev(action.payload ?? []);
+      return prevPage ? { ...state, page: prevPage } : state;
+
+    case "reset":
+      return { page: "RoutingInfo" };
+
+    case "goto":
+      const step = action.payload?.[0];
+      return step ? { ...state, page: step } : state;
+
+    default:
+      return state;
+  }
+}
+
+const Wizard = ({ isLoading = false, isError = false }) => {
+  const [state, dispatch] = useReducer(reducer, { page: "RoutingInfo" });
+  const form = useFormContext();
+  const methods = form.watch("methods");
```

Comment:
Suggest destructuring to leverage just those formContext pieces needed.  And leverage useWatch instead of watch to reduce overhead of rerenders

Reply:
Good point -- no real gain here